File: spring-boot-mybatis/src/main/java/com/example/app/controller/HomeController.java
package com.example.app.controller;

import com.example.app.domain.Users;
import com.example.app.service.UsersService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;

import java.util.List;

@Controller
public class HomeController {

    @Autowired
    private UsersService userService;

    @RequestMapping("/")
    public String home(Model model) {
        model.addAttribute("users", userService.findAll());
        return "home";
    }
}

File: spring-boot-kotlin-doma/src/main/kotlin/app/application/HomeController.kt
package app

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/")
class HomeController {
    @GetMapping
    fun home(model: Model): String {
        model.addAttribute("message", "Hello")
        return "home"
    }
}